test/index.test.ts

```diff
@@ -994,6 +994,155 @@ describe('HipThrusTS', () => {
         }
       });
     });
+    describe('sanitizers filtration functionality', () => {
```

Comment:
now here is only filtration functionality test
i was trying to refactor it like you say in https://github.com/captaincaius/hipthrusts/issues/21#issuecomment-631655727
but that approach have a little problems with types
when we have something like this: stage: AllStageKeys
return { [stage]: (context: { a: number }) => {
      expect(context.a).to.be.equal(constants.aPassedIn);
      return Promise.resolve({ b: constants.bPassedIn });
    }};
and we pass this function to HTPipe the return type of it would be {}
when from stage we removing all unions all works

Reply:
well the tests look good... hmm...
lemme think about what you're saying

Reply:
hmm I think I understand what you're saying... and I think I know why.  That function isn't generic so its return type won't be what we want 🤔

Reply:
```
function sanitizersFiltrationTestData<TStage extends AllStageKeys>(stage: TStage) {
  const testConstants = {
    aPassedIn: 'some string',
    bPassedIn: 'some other string',
  };
  const leftProjector = (context: { someObj: { a: string; b: string } }) => {
            expect(context).to.deep.equal({
              someObj: {
                a: testConstants.aPassedIn,
                b: testConstants.bPassedIn,
              },
            });
            return context.someObj;
          };
  const rightProjector = (context: { a: string; b: string }) {
            expect(context).to.not.has.property('someObj');
            expect(context).to.deep.equal({
              a: testConstants.aPassedIn,
              b: testConstants.bPassedIn,
            });
            return { b: context.b };
          }
  const testInput = 
          {
            someObj: {
              a: sanitizersFiltrationFunctionalityTestConstants.aPassedIn,
              b: sanitizersFiltrationFunctionalityTestConstants.bPassedIn,
            },
          };
  const testOutput = { b: sanitizersFiltrationFunctionalityTestConstants.bPassedIn };

  return {
    testConstants,
    left: {
      [stage]: leftProjector,
    }  as Record<TStage, typeof leftProjector>,
        right: {
      [stage]: rightProjector,
    }  as Record<TStage, typeof rightProjector>,
    testInput,
    testOutput,
  };
}
```
This also incorporates the other fix I wrote in the comment.

I'm not sure why typescript isn't smart enough to figure that out itself when I've seen it figure out WAY less obvious stuff, but I'm filing that in the tswtf category that I'll probably never figure out ;).
- [ ] guinea pig

Reply:
this is worked but...

Reply:
👍